### PR TITLE
Refactor payload

### PR DESCRIPTION
**Pull request by grillorafael**
Hello!

First of all thank you for this repo it is quite handy. It's a shame I don't work with mobile apps for a while.

I'm trying to add payload options but before I'm doing some refactoring around the modules.

I would like your review (this is still a work in progress because I didn't get to the payload part yet) moving forward.

About the payload, I thought about an option called `-p` that will be a payload literal `-p '{"postId: 10"}'` or ``-pf` to refer a json file.

Basically this PR at the moment introduces:

1. Add linter task to Gulp
1. Split code into smaller modules
1. Use promises around push
1. Rename config file in order to avoid conflict `config.json` is too generic

### CODE REVIEW

**Comment by kaplanmaxe** [comment]
Hey!

This is a great idea and looks great so far. I think that will be very handy, and something I should have thought of myself. Let me know when it's all set and I'll go ahead and merge it in. Thank you!

**Comment by grillorafael** [comment]
If you could review and maybe run some of the examples? I don't have gcm or apn setup here at the moment.

**Review by kaplanmaxe — COMMENTED** [review]
Everything looks good and seems to be working. Can you please just add docblocs to the functions?

File: src/utils/payload.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import fs from 'fs';
+
+export function loadPayloadFile(path) {
```
Comment (kaplanmaxe) [inline src/utils/payload.js]: doesn't look like you import this anywhere. Could you just make this a normal function?

**Comment by grillorafael** [comment]
About the docblocks, do you want it on which functions?

**Comment by grillorafael** [comment]
Also need to update npm version in order to update

**Comment by grillorafael** [comment]
Sorry for the PR mess I made now :)
